classes/services/PKPContextService.inc.php

```diff
@@ -485,6 +485,15 @@ public function edit($context, $params, $request) {
 			$params['styleSheet'] = $this->_saveFileParam($context, $params['styleSheet'], 'styleSheet', $userId);
 		}
 
+		$emailSignature = $context->getData('emailSignature');
+		foreach ($params as $key => $value) {
+			if (strpos($emailSignature, '{$' . $key . '}') !== false) {
+				$emailSignature = str_replace('{$' . $key . '}', $value, $emailSignature);
+			}
+		}
+
+		$context->setData('emailSignature', $emailSignature);
```

Comment:
Won't this get persisted when the context is saved later, effectively replacing `{$variableNames}` with variable values in the persisted version?

Reply:
The reason to do that is to display the signature correctly at `management/settings/workflow#emails` (but perhaps this is not the correct way)

Reply:
I understand, but this will probably affect things too broadly. I think the variable replacement should happen as close as possible to where it's used, so the variable replacements will be preserved without unexpected impacts...

With the old form toolset, this can be done in the setup form using the pkpTags plugin to TinyMCE (`plugins/generic/tinymce/plugins/pkpTags/plugin.js`, which provides support for displaying variables without replacing them with their values during editing. It's probably not supported in the new (vue.js-based) form toolset, though, and I think @NateWr probably has plans in this area.

The current examples of this in the codebase, e.g. the reviewer assignment process, are not great demonstrations of what I have in mind because they don't display the actual value of the content as the label on the variable, e.g.:
![image](https://user-images.githubusercontent.com/200411/108252532-9b5cdd00-710d-11eb-8167-4cede2948cad.png)

Reply:
It could be that the already existing default value also gets parsed and replaced (here)[https://github.com/pkp/pkp-lib/blob/master/classes/services/PKPSchemaService.inc.php#L449-L457] - So that, the `<a href=\"{$contextUrl}\">{$contextName}</a>` portion of the `default.contextSettings.emailSignature` is changed and inserted into the db like `<a href="http://localhost:8088/index.php/ts5">test_5</a>`
I think I followed the same principle here (or tried to) even though it may be a different case (?)

Reply:
Let's get @NateWr's opinion on this -- both in the way defaults are replaced vs. left for replacement on use in the SchemaDAO toolset, and on the use of variables in rich text controls. But generally I think variables that are not "known" at the time that code is run won't be affected since it comes down to a `str_replace('{$variableName}', 'variableValue', $default)`.

Reply:
OK, I removed the substitution code. Should I add something somewhere else (like @asmecher suggests [here](https://github.com/pkp/pkp-lib/pull/5154#discussion_r577859889)?)

Reply:
If you're working with one of the new forms, the `FieldRichTextarea` does support prepared content: https://docs.pkp.sfu.ca/dev/ui-library/dev/#/component/Form/fields/FieldRichTextarea/prepared-content. Look at the `preparedContent` and `renderPreparedContent` props in the docs there.

Reply:
Fixed (?)